### DISCUSSION TITLE
Is there a way to apply classes for *any* non-nullable value?

### DISCUSSION THREAD

**Discussion by lunelson** [joe-bell/cva#324] [Q&A]
Imagine you have a `Button` component, that takes optional `{ leftIcon?: Icon; rightIcon?: Icon: }` props. If either of these is supplied, you'd like to apply classes that modify the left and/or right padding of the button respectively. Is there any way to create a variants definition that can apply classes based on *any* non-nullable value being supplied to those props?

**Comment by GuyMevorachLevy** [comment]
Did you manage to find an answer? I'm working on that specific use-case aswell
